src/switcher.rs

```diff
@@ -45,18 +36,14 @@ pub fn switch_firmware(matches: &ArgMatches, paths: &ProjectDirs) -> Result<()>
     println!("Probe {} ({}) selected for firmware update", probe.firmware_identity()?, probe.serial_number()?);
 
     // Now extract the probe's identification, and check it's valid
-    let identity = parse_firmware_identity(&probe.firmware_identity()?);
-    let variant = identity.variant()?;
-    // If we don't know what version of firmware is on the probe, presume it's v1.6 for now..
-    // We can't actually know which prior version to v1.6 it actually is but it's very old either way
-    let firmware_version = identity.version.unwrap_or_else(|| "v1.6".into());
+    let identity: ProbeIdentity = probe.firmware_identity()?.into();
```

Comment:
This is the _only_ place where this presumption should be made, the parser for the firmware identity should **not** make this presumption as it does not generally hold. It only holds here.